File: app/src/main/java/de/jbamberger/irremote/service/ir/NECTranslator.java
package de.jbamberger.irremote.service.ir;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public class NECTranslator extends CodeTranslator {

    public NECTranslator() {
        super(new int[]{9000, 4500}, new int[]{560}, new int[]{560, 560}, new int[]{560, 1690}, 38000);
    }

    @Override
    public int[] buildCode(String codeString) {
        return buildRawCode(injectInverse(hexToBytes(codeString)));
    }
}

File: app/src/main/java/de/jbamberger/irremote/service/ir/NECTranslator.kt
package de.jbamberger.irremote.service.ir

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

internal class NECTranslator : CodeTranslator(
        initSequence = intArrayOf(9000, 4500),
        endSequence = intArrayOf(560),
        zero = intArrayOf(560, 560),
        one = intArrayOf(560, 1690),
        frequency = 38000) {

    override fun buildCode(codeString: String) =
            buildRawCode(injectInverse(hexToBytes(codeString)))

}
